searchcore/src/vespa/searchcore/proton/matching/sameelementmodifier.h

```diff
@@ -0,0 +1,35 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include "querynodes.h"
+#include <vespa/searchlib/query/tree/templatetermvisitor.h>
+
+namespace proton::matching {
+
+/**
+ * Prefix the indexname of the terms under the SameElement node.
+ *
+ */
+class SameElementModifier : public search::query::TemplateTermVisitor<SameElementModifier, ProtonNodeTypes>
+{
+public:
+    template <class TermNode>
+    void visitTerm(TermNode &) {  }
+
+    void visit(ProtonNodeTypes::SameElement &n) override {
+        if (n.getView().empty()) return;
+
+        vespalib::string prefix = n.getView() + ".";
+        for (auto & child : n.getChildren()) {
+            search::query::StringBase * term  = static_cast<search::query::StringBase *>(child);
```

Comment:
use Term class here (more generic). consider using dynamic cast to avoid crashing on unwanted input.

Reply:
I just saw it was a superclass of the class you used, which looked too specific. If we suddenly get support for searching with numbers, it will not work with same element...